File: src/main/kotlin/com/mikewerzen/automation/assistant/client/google/sheets/SheetsClient.java
package com.mikewerzen.automation.assistant.client.google.sheets;


import com.google.api.client.auth.oauth2.Credential;
import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
import com.google.api.client.http.HttpTransport;
import com.google.api.client.json.JsonFactory;
import com.google.api.client.json.jackson2.JacksonFactory;
import com.google.api.services.sheets.v4.Sheets;
import com.google.api.services.sheets.v4.model.ValueRange;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.List;

@Component
public class SheetsClient
{
	@Autowired
	private GoogleSecurityUtil googleSecurityUtil;

	private final JsonFactory JSON_FACTORY;

	private HttpTransport HTTP_TRANSPORT;

	@Value("${google.sheets.application.name}")
	private String applicationName;

	@Value("${google.sheets.spreadhseet.id}")
	private String spreadsheetId;


	public SheetsClient()
	{
		JSON_FACTORY = JacksonFactory.getDefaultInstance();

		try
		{
			HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
		} catch (Exception e)
		{
			throw new RuntimeException(e);
		}
	}

	public List<List<Object>> getDataInSheetRange(String sheetName, String range)
	{
		try
		{
			Sheets service = getSheetsService();

			String queryRange = sheetName + "!" + range;

			ValueRange response = service.spreadsheets().values()
					.get(spreadsheetId, queryRange)
					.execute();

			return response.getValues();

		} catch (Exception e)
		{
			throw new RuntimeException(e);
		}
	}

	private Sheets getSheetsService() throws IOException
	{
		Credential credential = googleSecurityUtil.authorize();
		return new Sheets.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
				.setApplicationName(applicationName)
				.build();
	}
}

File: kotlin/src/main/kotlin/com/mikewerzen/automation/assistant/client/google/sheets/SheetsClient.kt
package com.mikewerzen.automation.assistant.client.google.sheets


import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
import com.google.api.client.http.HttpTransport
import com.google.api.client.json.JsonFactory
import com.google.api.client.json.jackson2.JacksonFactory
import com.google.api.services.sheets.v4.Sheets
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class SheetsClient
{
	@Autowired
	private val googleSecurityUtil: GoogleSecurityUtil? = null

	private val JSON_FACTORY: JsonFactory

	private var HTTP_TRANSPORT: HttpTransport? = null

	@Value("\${google.sheets.application.name}")
	private val applicationName: String? = null

	@Value("\${google.sheets.spreadhseet.id}")
	private val spreadsheetId: String? = null

	init
	{
		JSON_FACTORY = JacksonFactory.getDefaultInstance()

		try
		{
			HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport()
		} catch (e: Exception)
		{
			throw RuntimeException(e)
		}

	}

	fun getDataInSheetRange(sheetName: String, range: String): List<List<Any>>
	{
		try
		{
			val service = sheetsService

			val queryRange = sheetName + "!" + range

			val response = service.spreadsheets().values()
					.get(spreadsheetId!!, queryRange)
					.execute()

			return response.getValues()

		} catch (e: Exception)
		{
			throw RuntimeException(e)
		}

	}

	private val sheetsService: Sheets
		@Throws(IOException::class)
		get()
		{
			val credential = googleSecurityUtil!!.authorize()
			return Sheets.Builder(HTTP_TRANSPORT!!, JSON_FACTORY, credential)
					.setApplicationName(applicationName)
					.build()
		}
}
